src/mem/alloc.h

```diff
@@ -62,20 +62,82 @@ namespace snmalloc
     Pagemap<SUPERSLAB_BITS, uint8_t, 0>>;
 
   HEADER_GLOBAL SuperslabPagemap global_pagemap;
+
+  /**
+   * Mixin used by `SuperslabMap` to directly access the pagemap via a global
+   * variable.  This should be used from within the library or program that
+   * owns the pagemap.
+   */
+  struct GlobalPagemap
+  {
+    /**
+     * Returns the pagemap.
+     */
+    SuperslabPagemap& pagemap()
+    {
+      return global_pagemap;
+    }
+  };
+
+  /**
```

Comment:
It might be more consistent to ifdef this section (including ExternalGlobalPagemap) using SNMALLOC_EXPOSE_PAGEMAP

Reply:
If we did ifdef that, we'd want to use a different macro, because one is for the compilation unit that exports the pagemap and the other is for the compilation units that import it.  In most use cases, consumers of this function will not be in libraries that implement it.  In the FreeBSD builds, we're exporting `snmalloc_pagemap_global_get` from libc, but we're then importing it into everything else that wants to interop with it.